src/hooks/useENSRegistrationCosts.ts

```diff
@@ -1,130 +1,119 @@
-    nativeCurrency
-  );
-
-  return {
-    amount,
-    display,
-    wei: networkFeeInWei,
-  };
-};
-
-const formatTotalRegistrationCost = (
-  wei: string,
-  nativeCurrency: any,
-  nativeAssetPrice: any
-) => {
-  const networkFeeInEth = fromWei(wei);
-
-  const { amount, display } = convertAmountAndPriceToNativeDisplay(
-    networkFeeInEth,
-    nativeAssetPrice,
-    nativeCurrency
-  );
-
-  return {
-    amount,
-    display,
-    wei,
-  };
-};
-
 export default function useENSRegistrationCosts({
   duration,
   name,
   rentPrice,
 }: {
   duration: number;
   name: string;
-  rentPrice: BigNumber;
+  rentPrice?: { perYear: { wei: number } };
 }) {
   const { nativeCurrency, accountAddress } = useAccountSettings();
 
-  const getRegistrationValuesEstimations = useCallback(async () => {
-    const nativeAssetPrice = ethereumUtils.getPriceOfNativeAssetForNetwork(
-      Network.mainnet
-    );
-    const gasLimit = await estimateENSRegistrationGasLimit(
-      name,
-      accountAddress,
-      duration * secsInYear,
-      rentPrice.toString()
-    );
-
-    const { gasFeeParamsBySpeed, currentBaseFee } = await getEIP1559GasParams();
-
-    const formattedEstimatedNetworkFee = formatEstimatedNetworkFee(
-      gasLimit,
-      currentBaseFee.gwei,
-      gasFeeParamsBySpeed.normal.maxPriorityFeePerGas.gwei,
-      nativeCurrency,
-      nativeAssetPrice
-    );
-
-    const weiEstimatedTotalCost = add(
-      formattedEstimatedNetworkFee.wei,
-      rentPrice.toString()
-    );
-    const totalRegistrationCost = formatTotalRegistrationCost(
-      weiEstimatedTotalCost,
-      nativeCurrency,
-      nativeAssetPrice
-    );
-
-    return {
-      estimatedTotalRegistrationCost: totalRegistrationCost,
-    };
-  }, [accountAddress, duration, name, nativeCurrency, rentPrice]);
+  const getEstimatedNetworkFee = useCallback(
+    async (_, { accountAddress, rentPricePerYear, name, nativeCurrency }) => {
+      const nativeAssetPrice = ethereumUtils.getPriceOfNativeAssetForNetwork(
+        Network.mainnet
+      );
+      const rentPrice = multiply(rentPricePerYear, duration);
+      const gasLimit = await estimateENSRegistrationGasLimit(
+        name,
+        accountAddress,
+        duration * secsInYear,
+        rentPrice
+      );
+
+      const {
+        gasFeeParamsBySpeed,
+        currentBaseFee,
+      } = await getEIP1559GasParams();
+
+      const formattedEstimatedNetworkFee = formatEstimatedNetworkFee(
+        gasLimit,
+        currentBaseFee.gwei,
+        gasFeeParamsBySpeed.normal.maxPriorityFeePerGas.gwei,
+        nativeCurrency,
+        nativeAssetPrice
+      );
+
+      return formattedEstimatedNetworkFee;
+    },
+    [duration]
+  );
 
-  const { data, status } = useQuery(
-    rentPrice && [
-      'getRegistrationValuesEstimations',
-      duration,
-      name,
-      rentPrice,
+  const rentPricePerYear = rentPrice?.perYear?.wei?.toString();
+  const { data: estimatedNetworkFee, status } = useQuery(
```

Comment:
Repurposed this `useQuery` to only fetch the estimated network fee, as the calculations of other costs are more reactive as the user can change the duration (we don't want the `useQuery` to be invoked on every duration change, as that could be a bit jarring).